src/main/java/nextstep/security/config/annotation/configurers/AuthorizeHttpRequestsConfigurer.java

```diff
@@ -0,0 +1,117 @@
+                ? context.getBean(RoleHierarchy.class) : new NullRoleHierarchy();
+    }
+
+    @Override
+    public void init(HttpSecurity http) {}
+
+    @Override
+    public void configure(HttpSecurity http) {
+        http.addFilter(new AuthorizationFilter(
+                registry.createAuthorizationManager()
+        ));
+    }
+
+    private AuthorizationManagerRequestMatcherRegistry addMapping(
+            List<? extends RequestMatcher> matchers, AuthorizationManager manager
+    ) {
+        for (RequestMatcher matcher : matchers) {
+            registry.addMapping(matcher, manager);
+        }
+        return registry;
+    }
+
+    public AuthorizationManagerRequestMatcherRegistry getRegistry() {
+        return registry;
+    }
+
+    public class AuthorizationManagerRequestMatcherRegistry {
+        private final List<RequestMatcherEntry<AuthorizationManager>> mappings = new ArrayList<>();
+
+        private void addMapping(RequestMatcher matcher, AuthorizationManager manager) {
+            mappings.add(new RequestMatcherEntry<>(matcher, manager));
+        }
+
+        private AuthorizationManager<HttpServletRequest> createAuthorizationManager() {
+            return new RequestMatcherDelegatingAuthorizationManager(mappings);
+        }
+
+        public AuthorizedUrl requestMatchers(String... patterns) {
+            return requestMatchers(null, patterns);
+        }
+
+        public AuthorizedUrl anyRequest() {
+            return requestMatchers(AnyRequestMatcher.INSTANCE);
+        }
+
+        public AuthorizedUrl requestMatchers(HttpMethod method, String... patterns) {
+            final List<RequestMatcher> matchers = new ArrayList<>();
+            for (String pattern : patterns) {
+                matchers.add(new MvcRequestMatcher(method, pattern));
+            }
+            return chainRequestMatchers(matchers);
+        }
+
+        public AuthorizedUrl requestMatchers(RequestMatcher... requestMatchers) {
+            return chainRequestMatchers(Arrays.asList(requestMatchers));
+        }
+
+        protected AuthorizedUrl chainRequestMatchers(List<RequestMatcher> requestMatchers) {
+            return new AuthorizedUrl(requestMatchers);
+        }
+    }
+
+    public class AuthorizedUrl {
+        private final List<? extends RequestMatcher> matchers;
+
+        AuthorizedUrl(List<? extends RequestMatcher> matchers) {
+            this.matchers = matchers;
+        }
+
+        public AuthorizationManagerRequestMatcherRegistry permitAll() {
+            return access((a, o) -> new AuthorizationDecision(true));
+        }
+
+        public AuthorizationManagerRequestMatcherRegistry hasRole(String role) {
+            return access(new AuthorityAuthorizationManager<>(roleHierarchy, role));
+        }
+
+        public AuthorizationManagerRequestMatcherRegistry authenticated() {
+            return access(new AuthenticatedAuthorizationManager<>());
+        }
```

Comment:
```suggestion
        private static final AuthorizationManager<?> PERMIT_ALL_MANAGER = (a, o) -> new AuthorizationDecision(true);
        private static final AuthorizationManager<?> AUTHENTICATED_MANAGER = new AuthenticatedAuthorizationManager<>();

        public AuthorizationManagerRequestMatcherRegistry permitAll() {
            return access(PERMIT_ALL_MANAGER);
        }

        public AuthorizationManagerRequestMatcherRegistry hasRole(String role) {
            return access(new AuthorityAuthorizationManager<>(roleHierarchy, role));
        }

        public AuthorizationManagerRequestMatcherRegistry authenticated() {
            return access(AUTHENTICATED_MANAGER);
        }
```

소소하지만.. 내부적으로 반복해서 생성될 필요 없는 객체는 미리 생성해둬도 좋겠네요!

Reply:
[5c28bc9](https://github.com/next-step/spring-security-refactoring/pull/1/commits/5c28bc9a1572a17d9236c352541dfb13f26944cc) 커밋에서 반영했습니다!